File: examples/example-58/js/features/support/mobile.js
const { Before, After } = require('@cucumber/cucumber');
const { remote } = require('webdriverio');
const config = require('config');
const path = require('path');

Before({ tags: '@mobile', timeout: 60000 }, async function () {
    const capabilities = config?.mobile?.appium?.capabilities;
    const appFile = path.join(process.cwd(), capabilities['appium:app']);
    capabilities['appium:app'] = appFile;

    this.mydemoapp = config?.mobile?.mydemoapp;
    this.appium = await remote({
        ...config?.mobile?.appium,
        capabilities
    });
});

After({ tags: '@mobile' }, async function () {
    if (this.appium) { await this.appium.deleteSession(); }
});

File: examples/example-58/ruby/features/support/mobile.rb
require 'appium_lib'

Before('@mobile') do
    proj = File.expand_path('../..', __dir__)
    caps = Settings.mobile.appium.to_h.deep_symbolize_keys
    caps[:caps][:app] = File.join(proj, Settings.mobile.appium.caps.app)
    @appium = Appium::Driver.new(caps, true)
    Appium.promote_appium_methods Object
end

After('@mobile') { @appium.driver_quit }

class MyDemoApp
    def initialize(appium)
        @appium = appium
    end

    def find_element(name)
        @appium.find_element(:xpath, Settings.mobile.mydemoapp[name])
    end
end